tests/end_to_end_tests/conftest.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2014-2022 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This is a root conftest for all pytest-based end-to-end tests. And it is mostly responsible for
+some common options and fixtures such as Scalyr credentials etc.
+"""
+
+import json
+import os
+import pathlib as pl
+import time
+
+import pytest
+from _pytest.runner import pytest_runtest_protocol as orig_pytest_runtest_protocol
+
+from agent_build.tools.constants import SOURCE_ROOT
+
+IN_CICD = os.environ.get("AGENT_BUILD_IN_CICD", False)
+
+
+def pytest_runtest_protocol(item, nextitem):
+    """
+    Wrap existing pytest protocol to print special grouping directive that
+    makes logs of each test case collapsable on GitHub Actions.
```

Comment:
One option would be to explicitly check for GHA here, but I think it's also fine since it's unlikely we will switch CI system in the near term.

Reply:
yes, I think we can change that whenever we want, for now if should be any big difference, since we are already putting out existing "IN_CI_CD" flag variable.